### PR TITLE
add keras2 MaxPooling2/3D layers and test

### PR DESCRIPTION
**Pull request by yjjiang11**

### CODE REVIEW

**Comment by yjjiang11** [comment]
Jenkins passed:http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/767/ @hkvision

**Comment by yjjiang11** [comment]
Jenkins passed: http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/772/ @hkvision

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 1]
```diff
@@ -172,3 +252,80 @@ def __init__(self,
...
+    name: String to set the name of the layer.
+          If not specified, its name will by default to be a generated string .
+
+    >>> maxpooling3d = MaxPooling3D((2, 1, 3), input_shape=(3, 32, 32, 32))
+    creating: createZooKeras2MaxPooling3D
+    """
+    def __init__(self,
+                 pool_size=(2, 2, 2),
+                 strides=None,
+                 data_format="channels_first",
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: where is padding?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras2.scala [hunk 2]
```diff
@@ -18,14 +18,10 @@ package com.intel.analytics.zoo.pipeline.api.keras.python
 
 import java.util.{List => JList}
 
-import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
 import com.intel.analytics.bigdl.optim.Regularizer
 import com.intel.analytics.bigdl.python.api.PythonBigDLKeras
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
-import com.intel.analytics.zoo.pipeline.api.Net
-import com.intel.analytics.zoo.pipeline.api.autograd.Variable
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras2.scala]: Why removing these lines?

File: zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/AveragePooling2DSpec.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      """.stripMargin
+    val seq = Sequential[Float]()
+    val layer = AveragePooling2D[Float](inputShape = Shape(3, 24, 24))
+    seq.add(layer)
+    seq.getOutputShape().toSingle().toArray should be (Array(-1, 3, 12, 12))
+    checkOutputAndGrad(seq.asInstanceOf[AbstractModule[Tensor[Float], Tensor[Float], Float]],
+      kerasCode)
+  }
+
+  "AveragePooling2D NHWC" should "be the same as Keras" in {
```
Comment (hkvision) [inline zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/AveragePooling2DSpec.scala]: why no taggedAs Keras2Test??

**Comment by hkvision** [comment]
Except for the above comments, others LGTM. Let's wait after 0.2 release to merge this :)

File: zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/AveragePooling2DSpec.scala (hunk 3 above)
Comment (zhichao-li) [reply zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/AveragePooling2DSpec.scala]: We should tagged this as keras2Test, and also make sure it would fail if without this
i.e add some keras2 specific parameters into the test.

Comment (hkvision) [reply zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/AveragePooling2DSpec.scala]: Then in this case he doesn't tag this as Keras2Test, why jenkins passed?

Comment (zhichao-li) [reply zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/AveragePooling2DSpec.scala]: if without such tag, then it would be tested against keras1.

Comment (hkvision) [reply zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/AveragePooling2DSpec.scala]: jenkins not passed http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/772/ 
@yjjiang11 please fix the unit tests.

**Comment by hkvision** [comment]
http://10.239.47.211:18888/job/ZOO-PR-Scala-Spark-1.6/796/ jenkins failed...

File: pyzoo/zoo/pipeline/api/keras2/layers/pooling.py [hunk 4]
```diff
@@ -172,3 +252,82 @@ def __init__(self,
...
+    creating: createZooKeras2MaxPooling3D
+    """
+    def __init__(self,
+                 pool_size=(2, 2, 2),
+                 strides=None,
+                 padding="valid",
+                 data_format="channels_first",
+                 input_shape=None,
+                 **kwargs):
+        super(MaxPooling3D, self).__init__(None,
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras2/layers/pooling.py]: add check for padding just like AveragePooling3D

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/MaxPooling3D.scala [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ *                be equal to poolSize.
+ * @param padding  Only 'valid' is supported new .
+ * @param dataFormat Format of input data. Please use "channels_first".
+ * @param inputShape A Single Shape, does not include the batch dimension.
+ * @tparam T Numeric type of parameter(e.g. weight, bias). Only support float/double now
+ */
+class MaxPooling3D[T: ClassTag](
+      override val poolSize: Array[Int] = Array(2, 2, 2),
+      override val strides: Array[Int] = null,
+      val padding: String = "valid",
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/MaxPooling3D.scala]: Is padding used?
